test/regression/targets/sc-53334.cc

```diff
@@ -0,0 +1,206 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * When run, this program will create a simple 2D sparse array, write some data
+ * to it in global order, and read the data back with aggregates.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <tiledb/tiledb.h>
+#include <tiledb/tiledb_experimental.h>
+#include <tiledb/tiledb>
+#include <vector>
+
+#include <catch2/catch_template_test_macros.hpp>
+
+using namespace tiledb;
+
+typedef char min_type;
+
+template <tiledb_datatype_t attribute_type>
+struct AttributeTraits;
+
+template <>
+struct AttributeTraits<TILEDB_UINT8> {
+  typedef uint8_t value_type;
+};
+
+template <>
+struct AttributeTraits<TILEDB_STRING_ASCII> {
+  typedef char value_type;
+};
+
+template <tiledb_datatype_t attribute_datatype>
+struct MyArray {
+  using AttributeValueType =
+      typename AttributeTraits<attribute_datatype>::value_type;
+
+  static void create_array(Context& ctx, const char* array_name) {
+    // The array will be 4x4 with dimensions "rows" and "cols", with domain
+    // [1,4].
+    auto rows = Dimension::create<int32_t>(ctx, "rows", {{1, 4}}, 4);
+    auto cols = Dimension::create<int32_t>(ctx, "columns", {{1, 4}}, 4);
+
+    // Create domain
+    Domain domain(ctx);
+    domain.add_dimension(rows);
+    domain.add_dimension(cols);
+
+    // Create a single attribute "a" so each (i,j) cell can store a character
+    Attribute a(ctx, "a", attribute_datatype);
+
+    // Create array schema
+    ArraySchema schema(ctx, TILEDB_SPARSE);
+    schema.set_domain(domain);
+    schema.set_cell_order(TILEDB_ROW_MAJOR);
+    schema.set_tile_order(TILEDB_ROW_MAJOR);
+    schema.add_attribute(a);
+
+    // Create array
+    Array::create(array_name, schema);
+  }
+
+  static void write_array(Context& ctx, const char* array_name) {
+    // Open array for writing
+    Array array(ctx, array_name, TILEDB_WRITE);
+
+    // Data
+    std::vector<int32_t> rows = {
+        1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4};
+    std::vector<int32_t> cols = {
+        1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4};
+    std::vector<AttributeValueType> atts = {
+        'a',
+        'b',
+        'c',
+        'd',
+        'e',
+        'f',
+        'g',
+        'h',
+        'i',
+        'j',
+        'k',
+        'l',
+        'm',
+        'n',
+        'o',
+        'p'};
+
+    Query query(ctx, array);
+    query.set_data_buffer("rows", rows)
+        .set_data_buffer("columns", cols)
+        .set_data_buffer("a", atts);
+
+    query.submit();
+    array.close();
+  }
+
+  static AttributeValueType query_min(Context& ctx, const char* array_name) {
+    // note, use C API because the CPP API doesn't seem to have Min yet
+    Array array(ctx, array_name, TILEDB_READ);
+
+    Query query(ctx, array);
+    query.set_layout(TILEDB_UNORDERED);
+
+    // use C API to request min() as it does not appear to be in CPP API yet
```

Comment:
It is in the CPP API. Look at TileDB/examples/cpp_api/aggregates.cc

Reply:
Ah, I see it now, not sure how I bungled this.  Though as you'll see in 3355c931 I did realize the crucial difference between `tiledb_experimental.h` and `tiledb_experimental` so maybe that's why I missed it.

Oh well, fixed now.